app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java

```diff
@@ -255,7 +255,16 @@ private void bindSelect(final StudyParameter parameter) {
         optionName.setText(parameter.heading);
         final TextView textView = (TextView) v.findViewById(R.id.value);
         selectView = textView;
-        textView.setText(String.valueOf(parameter.value));
+        String displayText = (String) parameter.value;
+        // Use the default value for Moving Average instead of mapping the value
+        if(parameter.heading.contains("Moving Average")) {
```

Comment:
Is there any time you would want to use parameter.value over parameter.defaultInput? Rather than checking for the hard string "Moving Average"

Reply:
We control they types of moving average on the charting library. The defaultValue will always be the key value in the moving average type map. Instead of creating a new HashMap with values we can just use the parameters.options hashmap that is returned from the library.